smartsim/builders/ensemble.py

```diff
@@ -163,7 +165,12 @@ def exe(self, value: str | os.PathLike[str]) -> None:
         """Set the executable.
 
         :param value: the executable
+        :raises TypeError: if the exe argument is not str or PathLike str
         """
+        if value:
+            if not isinstance(value, (str, os.PathLike)):
+                raise TypeError("exe argument was not of type str or PathLike str")
```

Comment:
Any reason for the `if value:` check here? Should we change it to just be
```suggestion
        if not isinstance(value, (str, os.PathLike)):
            raise TypeError("exe argument was not of type str or PathLike str")
```
so that a `TypeError` will still be raised in the even that someone does something silly like
```py
ens = Ensemble(...)
ens.exe = None  # <- `TypeError` now raised here
```
?

Reply:
changed